manager/integration/tests/test_rwx.py

```diff
@@ -1,12 +1,358 @@
+                   }
+                ],
+                "terminationGracePeriodSeconds": 30,
+                "tolerations": [
+                   {
+                      "effect": "NoExecute",
+                      "key": "node.kubernetes.io/not-ready",
+                      "operator": "Exists",
+                      "tolerationSeconds": 60
+                   },
+                   {
+                      "effect": "NoExecute",
+                      "key": "node.kubernetes.io/unreachable",
+                      "operator": "Exists",
+                      "tolerationSeconds": 60
+                   }
+                ]
+             }
+          }
+       }
+    }
+
+    def finalizer():
+        api = get_apps_api_client()
+        delete_and_wait_deployment(api, 'longhorn-nfs-provisioner',
+                                   namespace='longhorn-system')
+
+    request.addfinalizer(finalizer)
+
+    return deployment_nfs_manifest
+
+
+@pytest.fixture(scope="module", autouse="True")
+def nfs_pvc(request):
+    pvc_manifest = {
+        'apiVersion': 'v1',
+        'kind': 'PersistentVolumeClaim',
+        'metadata': {
+            'name': 'longhorn-nfs-provisioner',
+            'namespace': 'longhorn-system'
+        },
+        'spec': {
+            'storageClassName': 'longhorn',
+            'accessModes': [
+                'ReadWriteOnce'
+            ],
+            'resources': {
+                'requests': {
+                    'storage': '20G'
+                }
+            }
+        }
+    }
+
+    def finalizer():
+        api = k8sclient.CoreV1Api()
+        client = get_longhorn_api_client()
+        volumes = client.list_volume()
+
+        if not check_pvc_existence(api, pvc_manifest['metadata']['name'],
+                                   namespace='longhorn-system'):
+            return
+
+        volume_name = ''
+        for v in volumes:
+            print(v.kubernetesStatus.pvcName)
+            if v.kubernetesStatus.pvcName == 'longhorn-nfs-provisioner':
+                volume_name = v.name
+                client.delete(v)
+                break
+        wait_for_volume_delete(client, volume_name)
+
+        core_api = get_core_api_client()
+        delete_and_wait_pvc(core_api, pvc_manifest['metadata']['name'])
+
+    request.addfinalizer(finalizer)
+
+    return pvc_manifest
+
+
+@pytest.fixture(scope="module", autouse="True")
+def nfs_storage_class(request):
+    sc_manifest = {
+        'apiVersion': 'storage.k8s.io/v1',
+        'kind': 'StorageClass',
+        'metadata': {
+            'name': NFS_STORAGECLASS_NAME
+        },
+        'provisioner': 'nfs.longhorn.io',
+        'mountOptions': [
+            'vers=4.1',
+            'noresvport'
+        ]
+    }
+
+    def finalizer():
+        delete_storage_class(NFS_STORAGECLASS_NAME)
+
+    request.addfinalizer(finalizer)
+
+    return sc_manifest
+
+
+@pytest.fixture(scope="module", autouse="True")
+def core_api(request):
```

Comment:
We already have `core_api` fixture defined in `common.py`

Reply:
After discussing with @khushboo-rancher
`core_api` and `apps_api` defined in `common.py` is function scoped fixtures
It is pytest limitation that function scoped fixture can't be used in higher scoped fixtures

One minor change we can introduce, is to rename module scoped `core_api`  to something like `core_api_module` so it can be easily differentiated from function scoped `core_api` in `common.py`.  it's totally not to change it though.

Reply:
Changed the name to ```core_api_module```